TICodeView/Sources/Views/TICodeView/TICodeViewModel.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  Copyright (c) 2020 Touch Instinct
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the Software), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+
+/// Basic ViewModel of textFieldView for entering the verification code
+open class TICodeViewModel {
+    
+    public weak var delegate: TICodeViewModelDelegate?
+    
+    public var onTextEnter: ((String) -> Void)?
+    
+    public let codeConfig: TICodeConfig
+    
+    public init(codeConfig: TICodeConfig) {
+        self.codeConfig = codeConfig
+    }
+    
+    public func set(text: String) {
+        delegate?.update(text: text)
+    }
+    
+    public func set(focus: Bool) {
+        delegate?.update(focus: focus)
+    }
+    
+    public func clearText() {
+        set(text: "")
+    }
+}
```

Comment:
looks like useless proxy. you can move this code to view itself. library can have simpler API